tests/ui/needless_return.rs

```diff
@@ -365,4 +365,8 @@ fn conjunctive_blocks() -> String {
     return { "a".to_string() } + "b" + { "c" };
 }
 
+fn issue12907() -> String {
+    return "".split("").next().unwrap().to_string();
+}
+
```

Comment:
I think this just needs a test for the mentioned `RefCell` or any other type that fails to compile. but I think the significant drop check is adequate

There is the `rustc_insigificant_dtor` attribute. (I didn't even know this existed before checking.) This is given on the types `Rc`, `HashMap`, and `array::IntoIter`. Not sure whether this attribute changes the behavior (i.e., compiling would succeed) but it'd be a good idea to check.

Reply:
> I think this just needs a test for the mentioned `RefCell` or any other type that fails to compile. But I think the significant drop check is adequate

So should i delete the test then? 

> Not sure whether this attribute changes the behavior (i.e., compiling would succeed) but it'd be a good idea to check.

Yes, it compiles with the rustc_insigificant_dtor attribute and the lint is triggered.

Reply:
No I meant the only change it needs

Reply:
There is already a very similar test to the one mentioned:

``` rust
fn temporary_outlives_local() -> String {
    let x = RefCell::<String>::default();
    return x.borrow().clone();
}
```

Reply:
Yeah that does the same thing :+1: missed that before